packages/fgtclb/academic-projects/Classes/Domain/Model/Dto/ProjectDemand.php

```diff
@@ -55,14 +55,18 @@ public function getShowSelected(): bool
         return $this->showSelected;
     }
 
-    public function setHideCompletedProjects(bool $hideCompletedProjects): void
+    public function setActiveState(string $activeState): void
```

Comment:
We should use native PHP BackedEnums for ActiveState instead of string values again,
also for all related places.

Reply:
Thinking about that, or better to say during refactoring towards it it becomes clear that this is not reasonable.

`fluid` currently does not support native PHP Enums, baked or not baked, and cannot be used simply in fluid templates. The same issue has been introduced first as a possible regression recently in the TYPO3 core for FAL related stuff, but returned immediatly.

That means, we should introduce a PHP Enum only for internal use in the meantime and using good old strings in the DemandObject. With that, we can still use the new Enum as sanatized way on PHP handling code but keep compat and usage in fluid templates workable.